### PR TITLE
Convert main README to asciidoc

### PR DESCRIPTION
**Pull request by spinscale**
Some minor changes along the way

* Mentioning Slack instead of IRC
* Removed mentioning of native java API, HTTP should be used
* Removed java as a installed requirement

### CODE REVIEW

File: README.asciidoc [hunk 1]
```diff
@@ -15,39 +15,34 @@ Elasticsearch is a distributed RESTful search engine built for the cloud. Featur
...
 
-You need to have a recent version of Java installed. See the "Setup":http://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html#jvm-version page for more information.
-
-h3. Installation
-
-* "Download":https://www.elastic.co/downloads/elasticsearch and unzip the Elasticsearch official distribution.
-* Run @bin/elasticsearch@ on unix, or @bin\elasticsearch.bat@ on windows.
-* Run @curl -X GET http://localhost:9200/@.
+* https://www.elastic.co/downloads/elasticsearch[Download] and unpack the Elasticsearch official distribution.
+* Run `bin/elasticsearch` on unix, or `bin\elasticsearch.bat` on windows.
```
Comment (jrodewig) [inline README.asciidoc]: `Unix` and `Windows` should be capitalized. I also feel that distinctly calling out Linux or macOS is slightly more clear.
```suggestion
* Run `bin/elasticsearch` on Linux or macOS. Run `bin\elasticsearch.bat` on Windows.
```

**Review by jrodewig — APPROVED** [review]
I left one minor nit re: OS names but otherwise LGTM.

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix-unix

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix-windows

File: README.asciidoc (hunk 1 above)
Comment (nik9000) [reply README.asciidoc]: +1

File: README.asciidoc [hunk 2]
```diff
@@ -15,39 +15,34 @@ Elasticsearch is a distributed RESTful search engine built for the cloud. Featur
...
 * Start more servers ...
 
-h3. Indexing
+=== Indexing
 
-Let's try and index some twitter like information. First, let's index some tweets (the @twitter@ index will be created automatically):
+Let's try and index some twitter like information. First, let's index some tweets (the `twitter` index will be created automatically):
 
-<pre>
+-------------------------------------
```
Comment (nik9000) [inline README.asciidoc]: Code blocks only [need](https://asciidoctor.org/docs/user-manual/#built-in-blocks-summary) four dashes `----`. I think it'd be nice to go with that while we're here. No big deal, but it'd be nice.

**Review by nik9000 — COMMENTED** [review]
:+1:

File: README.asciidoc (hunk 2 above)
Comment (spinscale) [reply README.asciidoc]: done

File: README.asciidoc (hunk 1 above)
Comment (spinscale) [reply README.asciidoc]: done

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
I can't seem to get the `packaging-sample` CI tests to pass for this PR. However, I don't believe those tests are actually needed for these changes anyway.

I'll merge this. Thanks again @spinscale.

**Comment by jrodewig** [comment]
master:  56cb357 
7.x: 6fc7868485415e683c8cef4e0963257d3b321ab3
7.5: 548f209d32ee41abe7bca19b219e3ca80afabb81

**Pull request merged**